src/components/enter-mfa/enter-mfa-controller.ts

```diff
@@ -1,19 +1,68 @@
 import { Request, Response } from "express";
-import { NOTIFICATION_TYPE } from "../../app.constants";
+import {
+  NOTIFICATION_TYPE,
+  PATH_NAMES,
+  MFA_METHOD_TYPE,
+} from "../../app.constants";
 import { VerifyCodeInterface } from "../common/verify-code/types";
 import { codeService } from "../common/verify-code/verify-code-service";
 import { verifyCodePost } from "../common/verify-code/verify-code-controller";
 import { ExpressRouteFunc } from "../../types";
-import { ERROR_CODES } from "../common/constants";
+import { ERROR_CODES, pathWithQueryParam } from "../common/constants";
 import { supportAccountRecovery } from "../../config";
+import { AccountRecoveryInterface } from "../common/account-recovery/types";
+import { accountRecoveryService } from "../common/account-recovery/account-recovery-service";
+import { BadRequestError } from "../../utils/error";
 
 const TEMPLATE_NAME = "enter-mfa/index.njk";
 
-export function enterMfaGet(req: Request, res: Response): void {
-  res.render(TEMPLATE_NAME, {
-    phoneNumber: req.session.user.phoneNumber,
-    supportAccountRecovery: supportAccountRecovery() ? true : null,
-  });
+export function enterMfaGet(
+  service: AccountRecoveryInterface = accountRecoveryService()
+): ExpressRouteFunc {
+  return async function (req: Request, res: Response) {
+    const isAccountRecoveryEnabledForEnvironment = supportAccountRecovery();
+
+    if (!isAccountRecoveryEnabledForEnvironment) {
+      return res.render(TEMPLATE_NAME, {
+        phoneNumber: req.session.user.phoneNumber,
+        supportAccountRecovery: false,
+      });
+    }
+
+    const { email } = req.session.user;
+    const { sessionId, clientSessionId, persistentSessionId } = res.locals;
+
+    const accountRecoveryResponse = await service.accountRecovery(
+      sessionId,
+      clientSessionId,
+      email,
+      req.ip,
+      persistentSessionId
+    );
+
+    if (!accountRecoveryResponse.success) {
+      throw new BadRequestError(
+        accountRecoveryResponse.data.message,
+        accountRecoveryResponse.data.code
+      );
+    }
+
+    const isAccountRecoveryPermittedForUser =
+      (req.session.user.isAccountRecoveryPermitted =
+        accountRecoveryResponse.data.accountRecoveryPermitted);
+
+    const checkEmailLink = pathWithQueryParam(
+      PATH_NAMES.CHECK_YOUR_EMAIL_CHANGE_SECURITY_CODES,
+      "type",
+      MFA_METHOD_TYPE.SMS
+    );
+
+    res.render(TEMPLATE_NAME, {
+      phoneNumber: req.session.user.phoneNumber,
+      supportAccountRecovery: isAccountRecoveryPermittedForUser,
```

Comment:
I'll need to update this as the link always displays if account recovery is switched on.  If you're not allowed to do it it goes to the 'you cannot' page.

Reply:
Good point. Would you mind pulling latest and seeing if this works for you please?